Cargo.toml

```diff
@@ -14,7 +14,7 @@ members = [
 resolver = "2"
 
 [workspace.package]
-edition = "2021"
+edition = "2024"
```

Comment:
this is my first time encountering rust editions. Does this mean that our downstream consumers will require this edition too?

Reply:
no, unless our APIs use features exclusive on edition 2024, then consumers of our crate may have compatibility issues.